src/legacy/core_plugins/kibana/common/field_formats/types/string.js

```diff
@@ -57,6 +58,7 @@ export function createStringFormat(FieldFormat) {
         case 'title': return this._toTitleCase(val);
         case 'short': return shortenDottedString(val);
         case 'base64': return this._base64Decode(val);
+        case 'url': return decodeURIComponent(val);
```

Comment:
Thanks for your advice.

There two space strings '+' or '%20' in urlencoded.
decodeURI() and decodeURIComponent() can't decode '+' ,
so I fix with decodeURI(val.replace(/\+/g, ' '));

Please, check my commit.

Reply:
> There two space strings '+' or '%20' in urlencoded.
> decodeURI() and decodeURIComponent() can't decode '+' ,
> so I fix with decodeURI(val.replace(/+/g, ' '));

I'm not sure if `+` is something that needs to be included here -- `decodeURIComponent` is based on the URI spec, [which includes `+` as a reserved character](https://tools.ietf.org/html/rfc3986#section-2.2). That's why `encodeURIComponent` is going to encode it as `%2B`. 

We should be clear about exactly what type of data we are decoding with this transform. If we are focused on percent-encoded URL params, it makes sense to call it `urlparam` and decode purely with `decodeURIComponent`. That way it is clear that it is designed to handle anything encoded using `encodeURIComponent`.

If we wanted to handle data encoded with `encodeURI` instead -- which allows using `+` -- we should consider a different transform name and purely use `decodeURI`.

Personally, my opinion on this would be to call it `urlparam` as you have, and stick with using `decodeURIComponent` without modification. Would that still accomplish what you are looking to do?

Reply:
I didn't know that `+` was a reserved character in URI Spec. 
Okay, I fixed with `case 'urlparam': return decodeURIComponent(val);` and removed `+` converting from test case.
I tested `yarn test:mocha "src/**/field_formats/types/__tests__/**/string.js"` in my local environment and pushed.